### PR TITLE
kubeadm: update legacy kubelet config map information and document feature gates

### PR DESCRIPTION
**Pull request by neolit123**
has two separate commits.
- first removes some instances of the legacy naming (kubelet-config-x.yy), controller by the UnversionedKubeletConfigMap feature which becomes beta in 1.24 and is enabled by default.
- second documents the current state of feature gates

fixes https://github.com/kubernetes/website/issues/28788
xref https://github.com/kubernetes/kubeadm/issues/1582

### CODE REVIEW

**Comment by sftim** [comment]
/approve
/hold
I'm pre-approving this as the intent seems clear.

@neolit123 the /hold is only in case a stray LGTM comes in earlier than you'd hope for. It's fine to unhold this when this has LGTM and _you_ are happy to see this merge.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: updated.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: thanks for the edits. updated to use these, with one minor diff:
`kubeadm tool -> kubeadm` since kubeadm is the project name and is considered more of a toolkit

> I'd use a definition list

i have not used formatting such as:

```
`foo`
: text
```

but i guess i can check how that looks in the preview.
updated.

**Comment by neolit123** [comment]
/cc @SataQiu @RA489
PTAL for tech review / LGTM

**Comment by neolit123** [comment]
/cc @pacoxu 
as well since you have been following this effort.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (sftim) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: [preview](https://deploy-preview-31687--kubernetes-io-vnext-staging.netlify.app/docs/reference/setup-tools/kubeadm/kubeadm-init/#feature-gates)

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:155]: No need to use preformatted text here. Consider using an accessible table though - see https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#table-captions

Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: updated

File: content/en/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver.md
Comment (pacoxu) [inline content/en/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver.md:99]: I think we can remove this line as the name is defined.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (pacoxu) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:154]: UnversionedKubeletConfigMap
- 1.23 alpha
- 1.24 beta

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (pacoxu) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:134]: s/support/supports/g

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (SataQiu) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:166]:
```suggestion
for `kube-apiserver`, `kube-controller-manager`, `kube-scheduler` and `etcd` to run as non-root users.
```

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (SataQiu) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:175]:
```suggestion
that ConfigMap are appropriate for the value you set. When kubeadm writes this ConfigMap (during `kubeadm init`
```

File: content/en/docs/setup/production-environment/tools/kubeadm/kubelet-integration.md
Comment (SataQiu) [inline content/en/docs/setup/production-environment/tools/kubeadm/kubelet-integration.md:106]:
```suggestion
at `/var/lib/kubelet/config.yaml`, and also uploaded to a `kubelet-config` ConfigMap in the `kube-system` namespace of the cluster.
```

File: content/en/docs/setup/production-environment/tools/kubeadm/kubelet-integration.md
Comment (SataQiu) [inline content/en/docs/setup/production-environment/tools/kubeadm/kubelet-integration.md:107]:
```suggestion
A kubelet configuration file is also written to `/etc/kubernetes/kubelet.conf`
```

**Review by SataQiu — COMMENTED** [review]
LGTM, other than some nits!

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:154]: it might be better to make the table:

name | value | alpha | beta
------|-------|-------|-----
`foo` | `false` | 1.x | 1.y

**Comment by neolit123** [comment]
@SataQiu @pacoxu thanks for review.
updated.

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md
Comment (tengqm) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:158]: So we are pretty confident that no feature gate will be deprecated? All feature gates for kubeadm will eventually go GA?

Comment (sftim) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:158]: Good question.

Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:158]: If a FG is deprecated it can stay as alpha /beta in the table, but we should add a deprecated note in the description.

**Review by SataQiu — APPROVED** [review]
/lgtm

**Pull request merged**
